lib/rspec/core/configuration.rb

```diff
@@ -1412,13 +1415,32 @@ def self.delegate_to_ordering_manager(*methods)
       #     end
       #   end
       #
-      #   describe MyClass, :order => :reverse do
+      #   RSpec.describe 'MyClass', :order => :reverse do
       #     # ...
       #   end
       #
       # @note Pass the symbol `:global` to set the ordering strategy that
       #   will be used to order the top-level example groups and any example
       #   groups that do not have declared `:order` metadata.
+      #
+      # @example
+      #   RSpec.configure do |rspec|
+      #     rspec.register_ordering :global do |examples|
+      #       acceptance, other = examples.partition do |example|
+      #         example.metadata[:type] == :acceptance
```

Comment:
@e2 -- your feedback is very valuable and I thank you for it.  But I think comments on a PR that is focused only on improving the documentation of the existing API is not the best place for it: API changes are out of scope for this PR (as it's just about documenting what's already there) and will result in these comments no longer being surfaced in an open issue or PR once this PR is merged, essentially causing your valuable comments to be "lost".  (Not really lost, but no longer visible on an open thing).

In addition, engaging on these issues, and considering how it interacts with the rest of RSpec's features, takes a lot of time, and while I can't speak for @JonRowe, for me personally, my time working on RSpec is quite limited and I open a small PR like this one expecting it to breeze through code review and get merged without much back and forth.  When instead a whole conversation is opened up that requires a _lot_ of time to adequately respond to, well, that's time I often don't have and it can be a bit frustrating to see a small PR get side tracked on a larger discussion that's really not in scope for the PR.  At the same time, I don't want to ignore valuable feedback like yours, so I make time to respond but it turns what was initially a quick drive-by contribution into a larger thing that I didn't really have time for this week.

Instead, if you opened an issue with this feedback, I think it would be much more productive for all involved.  It wouldn't sidetrack this small PR, and I'd participate knowing it's a longer conversation that's going to take some time.  It would also ensure this feedback is not "lost" when this PR is merged.

So I'd say please _do_ continue giving your valuable feedback, but please try to find the most appropriate place for it (often an issue if it's not in scope for a specific PR).

For this specific case, before you open an issue, I'll provide you with a few thoughts of mine on your feedback.

* We used to have a larger ordering API (with separate methods to order groups and examples) in RSpec 2.  However, the consensus was that it was unnecessarily confusing and it was simpler to consolidate to one method.  See #1025: there are extended conversations there about many of the issues you've mentioned. #1146 contains the deprecations of the old APIs (which should give you an idea of what the API used to be like).
* It's not clear to me how we could support the `<=>` API like Ruby's sort does.  I'd like to see a concrete proposal for how you would like the API to work differently.
* You mention that you might create a plugin but that API improvements on our end could render your plugin obselete.  It might actually make sense to first implement your ideas as a plugin, with the plan to later integrate it into RSpec (that's often how we've done things), although if you propose something that we're ready to support immediately and you're willing to do the implementation work we can probably bypass the plugin.  It all depends on what API you have in mind, which I really have no idea about.

Thanks!